src/Symfony/Component/Asset/VersionStrategy/JsonManifestVersionStrategy.php

```diff
@@ -46,23 +51,26 @@ public function getVersion(string $path)
     }
 
     public function applyVersion(string $path)
-    {
-        return $this->getManifestPath($path) ?: $path;
-    }
-
-    private function getManifestPath(string $path): ?string
     {
         if (null === $this->manifestData) {
             if (!is_file($this->manifestPath)) {
-                throw new \RuntimeException(sprintf('Asset manifest file "%s" does not exist.', $this->manifestPath));
+                throw new RuntimeException(sprintf('Asset manifest file "%s" does not exist.', $this->manifestPath));
             }
 
             $this->manifestData = json_decode(file_get_contents($this->manifestPath), true);
             if (0 < json_last_error()) {
-                throw new \RuntimeException(sprintf('Error parsing JSON from asset manifest file "%s": ', $this->manifestPath).json_last_error_msg());
+                throw new RuntimeException(sprintf('Error parsing JSON from asset manifest file "%s": ', $this->manifestPath).json_last_error_msg());
             }
         }
 
-        return isset($this->manifestData[$path]) ? $this->manifestData[$path] : null;
+        if (isset($this->manifestData[$path])) {
+            return $this->manifestData[$path];
+        }
+
+        if ($this->strict) {
+            throw new RuntimeException(sprintf('Asset "%s" not found in manifest "%s".', $path, $this->manifestPath));
```

Comment:
It may be overkill... but it would be kind of awesome if we can suggest some paths. Often, I see people barely have the wrong path - like bad capitalization or they miss the word “builds” or they have an extra “/“ in there.

I guess we would probably have to grab the actual filename part from the path (for example, from builds/foo.png we would grab foo.png) then see if any keys in the real manifest *end* in this string (both normalized for “case”). WDYT?

Reply:
Here it is. I took inspiration from [`Symfony\Console\Application::findAlternatives()`](https://github.com/symfony/symfony/blob/5.x/src/Symfony/Component/Console/Application.php#L1086-L1126)